src/audio/spectrogram.jl

```diff
@@ -0,0 +1,89 @@
+"""
+
+Create a spectrogram or a batch of spectrograms from a raw audio signal.
+
+# Arguments
+
+- `pad::Int`:
+    Then amount of padding to apply on both sides.
+    Default is `0`.
+- `window_normalized::Bool`:
+    Whether to normalize the waveform by the window’s L2 energy.
+    Default is `false`.
+- `power::Real`:
+    Exponent for the magnitude spectrogram (must be ≥ 0)
+    e.g., `1` for magnitude, `2` for power, etc.
+    If `0`, complex spectrum is returned instead.
+
+See [`stft`](@ref) for other arguments.
+
+# Returns
+
+Spectrogram in the shape `(T, F, B)`, where
+`T` is the number of window hops and `F = n_fft ÷ 2 + 1`.
+
+# Example
+
+```julia
+julia> waveform, sampling_rate = load("test.flac");
+
+julia> spec = spectrogram(waveform;
+    n_fft=1024, hop_length=128, window=hann_window(1024));
+
+julia> spec_db = NNlib.power_to_db(spec);
+
+julia> Makie.heatmap(spec_db[:, :, 1])
```

Comment:
See output for UnicodePlot for heatmap: https://fluxml.ai/NNlib.jl/previews/PR587/audio/#Spectrogram

Not sure how I feel about it. Maybe it is worth switching to Plots.jl?

Reply:
See output for UnicodePlot for heatmap: https://fluxml.ai/NNlib.jl/previews/PR587/audio/#Spectrogram

Not sure how I feel about it. Maybe it is worth switching to Plots.jl?

Reply:
Yeah it's not ideal, but my inclination towards UnicodePlots is so that docstrings render completely in the REPL as well. I'm happy to use Makie or Plots instead though.


Let's solicit opinions from @mcabbott or @ToucheSir.

Reply:
I've updated to use Makie plots, here's UnicodePlots for reference:

|UnicodePlots|Makie|
|-|-|
|![image](https://github.com/FluxML/NNlib.jl/assets/17990405/e766aaa4-3168-4bb9-bb2d-5567cf13736e)|![image](https://github.com/FluxML/NNlib.jl/assets/17990405/6e962982-d328-4e5f-abfb-3953ac302861)|